homeassistant/components/roborock/image.py

```diff
@@ -155,31 +175,54 @@ async def create_coordinator_maps(
     maps_info = sorted(
         coord.maps.items(), key=lambda data: data[0] == cur_map, reverse=True
     )
-    for map_flag, map_info in maps_info:
-        # Load the map - so we can access it with get_map_v1
-        if map_flag != cur_map:
-            # Only change the map and sleep if we have multiple maps.
-            await coord.api.send_command(RoborockCommand.LOAD_MULTI_MAP, [map_flag])
-            coord.current_map = map_flag
-            # We cannot get the map until the roborock servers fully process the
-            # map change.
-            await asyncio.sleep(MAP_SLEEP)
-        # Get the map data
-        map_update = await asyncio.gather(
-            *[coord.cloud_api.get_map_v1(), coord.get_rooms()], return_exceptions=True
+    maps = await hass.async_add_executor_job(
+        roborock_storage.exec_load_maps,
+        [roborock_map.name for roborock_map in coord.maps.values()],
+    )
+    storage_updates: list[tuple[str, bytes]] = []
+    for (map_flag, map_info), storage_map in zip(maps_info, maps, strict=False):
```

Comment:
The work below is executing sequential, but it also sleeps, which blocks all following work. Is it possible to wrap these and do with `asyncio.gather` ?

Reply:
I don't think so but let me give you some context and you can tell me if i'm missing something.

Most commands for roborock - I send the command - and then I get a response with the result, i.e. get_status will then respond back with the status.

set_multi_map on the other hand - returns 'ok' - and it sends ok as soon as it receieves the message. However, it takes a few seconds for the map to actually be loaded in - if you go too quickly, you will end up executing commands on the old map. Hence the sleep. I can't really group the sleep and the map change, as I can't sleep before I get the 'ok' back as there's a chance if there was a network slowdown, that it took longer than expected for the change map command to be sent. and I am then not sleeping an adequate amount of time.

This is the worst part of the api to work with, and is something I have tried multiple times to get assistance from anyone within Roborock for(with no luck).